src/components/AutomationDetails/index.jsx

```diff
@@ -1,126 +1,181 @@
+      </div>
+    );
+  }
+
+  renderSlackTitles = () => (
+    <div className="slack-table">
+      <div className="content-title">Channel</div>
+      <div className="content-title">Action</div>
+      <div className="content-title">Status</div>
+    </div>
+  );
+
+  renderEmailTitles = () => (
+    <div className="slack-table">
+      <div className="content-title">Recipient</div>
+      <div className="content-title">Subject</div>
+      <div className="content-title">Status</div>
+    </div>
+  );
+
+  renderFreckleTitles = () => (
+    <div className="slack-table">
+      <div className="content-title">Project Tag</div>
+      <div className="content-title">Action</div>
+      <div className="content-title">Status</div>
+    </div>
+  );
+
+  renderDetails(modalType, modalContent) {
     return (
-      <React.Fragment>
-        <td>{automationMedia}</td>
-        <td>{automationTitle}</td>
-        {status === 'success' ? <td>Success</td> : <td>Failed</td>}
-      </React.Fragment>
+      <div className="content-container">
+        {modalType === 'slack' && this.renderSlackDetails(modalContent)}
+        {modalType === 'email' && this.renderEmailDetails(modalContent)}
+        {modalType === 'freckle' && this.renderFreckleDetails(modalContent)}
+      </div>
     );
   }
 
+  renderSlackDetails = (modalContent) => {
+    const { slackAutomations } = modalContent;
+    const slackActivities = (slackAutomations && slackAutomations.slackActivities) || [];
+    return (slackActivities.map(content => (
+      <div key={content.id}>
+        <div className="automation-content">
+          <div className="content-row name">{content.channelName}</div>
+          <div className="content-row type">{content.type}</div>
+          {this.renderCommonDetails(content)}
+        </div>
+      </div>
+    )));
+  }
+
+  renderEmailDetails = (modalContent) => {
+    const { emailAutomations } = modalContent;
+    const emailActivities = (emailAutomations && emailAutomations.emailActivities) || [];
+    return (emailActivities.map(content => (
+      <div key={content.id}>
+        <div className="automation-content">
+          <div className="content-row name">{content.emailTo}</div>
+          <div className="content-row subject">{content.subject}</div>
+          {this.renderCommonDetails(content)}
+        </div>
+      </div>
+    )));
+  }
+
+  renderFreckleDetails = (modalContent) => {
+    const { freckleAutomations } = modalContent;
+    const freckleActivities = (freckleAutomations && freckleAutomations.freckleActivities) || [];
+    return (freckleActivities.map(content => (
+      <div key={content.id}>
+        <div className="automation-content">
+          <div className="content-row name">{content.projectId}</div>
+          <div className="content-row type">{content.type}</div>
+          {this.renderCommonDetails(content)}
+        </div>
+      </div>
+    )));
+  }
+
+  renderCommonDetails = content => (
+    <>
```

Comment:
How about you use `<React.Fragment></React.Fragment>` instead of the empty <></>?`

Reply:
Noted!